### PR TITLE
goto-analyzer: default result report now uses messaget

### PR DESCRIPTION
**Pull request by kroening**
Splitting 'report into text file' and 'report onto console' enables tweaking
the latter, e.g., by adding colors.

- [X] Each commit message has a non-empty body, explaining why the change was made.
- [X] My contribution is formatted in line with CODING_STANDARD.md.
- [X] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [X] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: e8e0cffd090e4a22f72d7c02d3e27bb0c8e20a82).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85660560

File: src/goto-analyzer/goto_analyzer_parse_options.cpp [hunk 1]
```diff
@@ -589,11 +584,13 @@ int goto_analyzer_parse_optionst::perform_analysis(const optionst &options)
 
     // Output file factory
     const std::string outfile=options.get_option("outfile");
+
     std::ofstream output_stream;
-    if(!(outfile=="-"))
+    if(outfile != "-" && outfile != "")
```
Comment (tautschnig) [inline src/goto-analyzer/goto_analyzer_parse_options.cpp]: I'd use `!outfile.empty()`

File: src/goto-analyzer/goto_analyzer_parse_options.cpp [hunk 2]
```diff
@@ -589,11 +584,13 @@ int goto_analyzer_parse_optionst::perform_analysis(const optionst &options)
...
       output_stream.open(outfile);
 
-    std::ostream &out((outfile=="-") ? std::cout : output_stream);
+    std::ostream &out(
+      (outfile == "-" || outfile == "") ? std::cout : output_stream);
```
Comment (tautschnig) [inline src/goto-analyzer/goto_analyzer_parse_options.cpp]: `outfile.empty()`

File: src/goto-analyzer/static_verifier.cpp [hunk 3]
```diff
@@ -143,10 +143,8 @@ bool static_verifier(
     m.status() << "Writing XML report" << messaget::eom;
     out << xml_result;
   }
-  else
+  else if(options.get_bool_option("text"))
```
Comment (tautschnig) [inline src/goto-analyzer/static_verifier.cpp:146]: Is this still being set anywhere?

Comment (peterschrammel) [reply src/goto-analyzer/static_verifier.cpp:146]: Looks obsolete now

**Review by peterschrammel — APPROVED** [review]
modulo Michael's comments

File: src/goto-analyzer/goto_analyzer_parse_options.cpp (hunk 1 above)
Comment (kroening) [reply src/goto-analyzer/goto_analyzer_parse_options.cpp]: Done

File: src/goto-analyzer/static_verifier.cpp (hunk 3 above)
Comment (kroening) [reply src/goto-analyzer/static_verifier.cpp:146]: Is set with command line option --text, as before.

Comment (tautschnig) [reply src/goto-analyzer/static_verifier.cpp:146]: Oh, it's just no longer the default. Ok!

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 2c19624929a5a7753acbc4cd6d4e4d9cc39dc803).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85832591

**Pull request merged**